devito/ir/support/space.py

```diff
@@ -148,9 +150,57 @@ def _rebuild(self):
 
     @cached_property
     def size(self):
-        upper_extreme = self.dim.symbolic_max + self.upper
-        lower_extreme = self.dim.symbolic_min + self.lower
-        return (upper_extreme - lower_extreme + 1) / self.dim.symbolic_incr
+        """
+        The Interval size, defined as the number of points iterated over through
+        ``self.dim``, namely
+
+            (dim.symbolic_max + upper - dim.symbolic_min - lower + 1) / dim.symbolic_incr
+
+        Notes
+        -----
+        The Interval size is typically a function of several symbols (e.g.,
+        `self.dim.symbolic_max`), and all such symbols must be mappable
+        to actual numbers at `op.apply` time (i.e., the runtime values).
+        When `self.dim` is an "unstructured Dimension", such as ModuloDimension,
+        things can get nasty since the symbolic min/max/incr can literally be
+        anything (any expression involving any Dimension/symbol/...), which makes
+        it extremely complicated to numerically compute the size. However,
+        the compiler only uses such unstructured Dimensions in well defined
+        circumstances, which we explicitly handle here. Ultimately, therefore,
+        we return a `size` that is made up of known symbols.
+        """
```

Comment:
So no case where we can end up here wit ha `Modulo/is_Conditional Dimension` right? WHat if one has a Subdimension of a modulo dimension?

Reply:
> So no case where we can end up here wit ha Modulo/is_Conditional Dimension right? 

correct. The possible cases are constrained by our user API ie IOW by what the user can write

> WHat if one has a Subdimension of a modulo dimension?

ModuloDimension is not part of our user API so we're good on this front ...

... but part of our API are SteppingDimension and SubDimension, and yeah, one could compose them and do nasty things, but I'm not even sure we would be able to generate anything functional, so...

at the end of the day, I'm just _enhancing_ this method, not dropping stuff, so at worst I'm extending coverage